test/unit-test/sigv4_utest.c

```diff
@@ -214,3 +493,209 @@ void test_SigV4_AwsIotDateToIso8601_Formatting_Error()
+    /* The attempt to encode the query field causes OOM (out of memory). */
+    resetInputParams();
+    params.pHttpParameters->pQuery = QUERY_ENCODE_FIELD_OOM;
+    params.pHttpParameters->queryLen = STR_LIT_LEN( QUERY_ENCODE_FIELD_OOM );
+    returnStatus = SigV4_GenerateHTTPAuthorization( &params, authBuf, &authBufLen, &signature, &signatureLen );
+    TEST_ASSERT_EQUAL( SigV4InsufficientMemory, returnStatus );
+
+    /* The attempt to encode the query value causes OOM (out of memory). */
+    resetInputParams();
+    params.pHttpParameters->pQuery = QUERY_ENCODE_VALUE_OOM;
+    params.pHttpParameters->queryLen = STR_LIT_LEN( QUERY_ENCODE_VALUE_OOM );
+    returnStatus = SigV4_GenerateHTTPAuthorization( &params, authBuf, &authBufLen, &signature, &signatureLen );
+    TEST_ASSERT_EQUAL( SigV4InsufficientMemory, returnStatus );
+
+    /* The attempt to write the '=' character before a value causes OOM (out of memory). */
+    resetInputParams();
+    params.pHttpParameters->pQuery = QUERY_EQUAL_BEFORE_VALUE_OOM;
+    params.pHttpParameters->queryLen = STR_LIT_LEN( QUERY_EQUAL_BEFORE_VALUE_OOM );
+    returnStatus = SigV4_GenerateHTTPAuthorization( &params, authBuf, &authBufLen, &signature, &signatureLen );
+    TEST_ASSERT_EQUAL( SigV4InsufficientMemory, returnStatus );
+
+    /* The attempt to write the '&' character before a field causes OOM (out of memory). */
+    resetInputParams();
+    params.pHttpParameters->pQuery = QUERY_AMPERSAND_BEFORE_FIELD_OOM;
+    params.pHttpParameters->queryLen = STR_LIT_LEN( QUERY_AMPERSAND_BEFORE_FIELD_OOM );
+    returnStatus = SigV4_GenerateHTTPAuthorization( &params, authBuf, &authBufLen, &signature, &signatureLen );
+    TEST_ASSERT_EQUAL( SigV4InsufficientMemory, returnStatus );
+    /* END: Coverage for writeCanonicalQueryParameters(). */
+
+    /* BEGIN: Coverage for writeLineToCanonicalRequest(). */
+    /* Writing a precanonicalized query to the processing buffer causes OOM. */
+    resetInputParams();
+    params.pHttpParameters->pQuery = PRECANON_QUERY_TOO_LONG;
+    params.pHttpParameters->queryLen = STR_LIT_LEN( PRECANON_QUERY_TOO_LONG );
+    params.pHttpParameters->flags = SIGV4_HTTP_QUERY_IS_CANONICAL_FLAG;
+    returnStatus = SigV4_GenerateHTTPAuthorization( &params, authBuf, &authBufLen, &signature, &signatureLen );
+    TEST_ASSERT_EQUAL( SigV4InsufficientMemory, returnStatus );
+    /* END: Coverage for writeLineToCanonicalRequest(). */
+}
+
+/**
+ * @brief Coverage for cases when the hashing functions return errors.
+ */
+void test_SigV4_GenerateHTTPAuthorization_Hash_Errors()
+{
+    SigV4Status_t returnStatus;
+    size_t i;
+
+    for( i = 0U; i < HAPPY_PATH_HASH_ITERATIONS; i++ )
+    {
+        resetFailableHashParams();
+        initHashCallToFail = i;
+        returnStatus = SigV4_GenerateHTTPAuthorization( &params, authBuf, &authBufLen, &signature, &signatureLen );
+        TEST_ASSERT_EQUAL( SigV4HashError, returnStatus );
```

Comment:
Would it be useful to know which iteration failed? Perhaps by using `TEST_ASSERT_EQUAL_MESSAGE` to log the failing iteration?

Reply:
Added